CIP-DappConnector/CIP-dapp-connector.md

```diff
@@ -0,0 +1,250 @@
+---
+CIP: 30
+Title: Cardano dApp-Wallet Web Bridge
+Authors: rooooooooob
+Comments-URI: https://github.com/cardano-foundation/CIPs/pull/88
+Status: Draft
+Type: Standards
+Created: 2021-04-29
+License: CC-BY-4.0
+---
+
+# Abstract
+
+This documents describes a webpage-based communication bridge allowing webpages (i.e. dApps) to interface with Cardano wallets. This is done via injected javascript code into webpages. This specification defines the manner that such code is to be accessed by the webpage/dApp, as well as defining the API for dApps to communicate with the user's wallet. This document currently concerns the Shelley-Mary era but will have a second version once Plutus is supported. This specification is intended to cover similar use cases as web3 for Ethereum or [EIP-0012](https://github.com/ergoplatform/eips/pull/23) for Ergo. The design of this spec was based on the latter.
+
+
+# Motivation
+
+In order to facilitate future dApp development, we will need a way for dApps to communicate with the user's wallet. While Cardano does not yet support smart contracts, there are still various use cases for this, such as NFT management. This will also lay the groundwork for an updated version of the spec once the Alonzo hardfork is released which can extend it to allow for Plutus support.
+
+
+
+# Specification
+
+## Version
+
+The API specified in this document will count as version 0.1.0 for version-checking purposes below.
+
+## Data Types
+
+### Bytes
+
+A hex-encoded string of the corresponding bytes.
+
+### cbor\<T>
+
+A hex-encoded string representing [CBOR](https://tools.ietf.org/html/rfc7049) corresponding to `T` defined via [CDDL](https://tools.ietf.org/html/rfc8610) either inside of the [Shelley Mult-asset binary spec](https://github.com/input-output-hk/cardano-ledger-specs/blob/0738804155245062f05e2f355fadd1d16f04cd56/shelley-ma/shelley-ma-test/cddl-files/shelley-ma.cddl) or, if not present there, from the [CIP-0008 signing spec](https://github.com/cardano-foundation/CIPs/blob/master/CIP-0008/CIP-0008.md).
```

Comment:
Why is the CBOR<T> represented as a hex string instead of a `Uint8Array`? Doesn't this representation incur in serialization/deserialization overhead?

Reply:
Yes it incurs serialization overhead, but that should be quite minimal compared to everything else going on. I do also like that we are maintaining that the API as pure and direct JSON compatible in case we ever want to extend it to other approaches (e.g. Wallet Connect like functionality).